packages/flutter/lib/src/widgets/lazy_block.dart

```diff
@@ -597,7 +600,10 @@ class _LazyBlockElement extends RenderObjectElement {
   void performRebuild() {
     IndexedWidgetBuilder builder = widget.delegate.buildItem;
     List<Widget> widgets = <Widget>[];
-    for (int i = 0; i < _children.length; i += 1) {
+    // If the most recent layout didn't fill the viewport but an additional child
+    // is now available, add it to the widgets list which will force a layout.
+    int buildChildCount = _underflow ? _children.length + 1 : _children.length;
```

Comment:
I actually removed the final qualifier because most of the code in this file doesn't use it for local vars (that are in fact final).

Would be happy to make another pass and finalize things.